src/components/Accordion/AccordionSummary/AccordionSummary.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+
+import {KeyCode} from '../../../constants';
+import {Disclosure} from '../../Disclosure';
+import type {DisclosureSummaryRenderFunctionProps} from '../../Disclosure/DisclosureSummary/DisclosureSummary';
+import type {QAProps} from '../../types';
+import {isOfType} from '../../utils/isOfType';
+import {useAccordionAttributes} from '../AccordionContext';
+import {accordionSummaryBlock} from '../constants';
+
+import './AccordionSummary.scss';
+
+type AccordionSummaryProps = QAProps & {
+    children: (
+        props: DisclosureSummaryRenderFunctionProps,
+        defaultSummary: React.ReactElement,
+    ) => React.ReactElement;
+};
+
+export function AccordionSummary(props: AccordionSummaryProps) {
+    const {children, qa} = props;
+    const attributes = useAccordionAttributes();
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+        if (
+            e.key === KeyCode.ARROW_DOWN ||
+            e.key === KeyCode.ARROW_UP ||
+            e.key === KeyCode.HOME ||
+            e.key === KeyCode.END
+        ) {
+            e.preventDefault();
+            const accordionContainer = e.currentTarget.closest('.g-accordion');
```

Comment:
But here (and below) it seems to be a good place to use context - you can put links to all the necessary refs in it and avoid using the native dom api